includes/server/class-llms-rest-memberships-controller.php

```diff
@@ -0,0 +1,590 @@
+			$prepared_item['sales_page_content_url'] = $request['sales_page_url'];
+		}
+
+		return $prepared_item;
+
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param LLMS_Membership $membership LLMS Membership.
+	 * @return array Links for the given object.
+	 */
+	protected function prepare_links( $membership ) {
+		$parent_links = parent::prepare_links( $membership );
+		$id           = $membership->get( 'id' );
+
+		$links = array();
+
+		// Enrollments.
+		$links['enrollments'] = array(
+			'href' => rest_url( sprintf( '/%s/%s/%d/%s', $this->namespace, $this->rest_base, $id, 'enrollments' ) ),
+		);
+
+		// Instructors.
+		$links['instructors'] = array(
+			'href' => add_query_arg(
+				'post',
+				$id,
+				rest_url( sprintf( '%s/%s', 'llms/v1', 'instructors' ) )
+			),
+		);
+
+		// Students.
+		$links['students'] = array(
+			'href' => add_query_arg(
+				'enrolled_in',
+				$id,
+				rest_url( sprintf( '%s/%s', 'llms/v1', 'students' ) )
+			),
+		);
+
+		return array_merge( $parent_links, $links );
+	}
+
+	/**
+	 * Prepare a single object output for response.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Membership $membership Membership object.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_object_for_response( $membership, $request ) {
+		$data    = parent::prepare_object_for_response( $membership, $request );
+		$context = $request->get_param( 'context' );
+
+		// Auto enroll.
+		$data['auto_enroll'] = $membership->get_auto_enroll_courses();
+
+		// Catalog visibility.
+		$data['catalog_visibility'] = $membership->get_product()->get_catalog_visibility();
+
+		// Categories.
+		$data['categories'] = $membership->get_categories(
+			array(
+				'fields' => 'ids',
+			)
+		);
+
+		// Instructors.
+		$instructors         = $membership->get_instructors();
+		$instructors         = empty( $instructors ) ? array() : wp_list_pluck( $instructors, 'id' );
+		$data['instructors'] = $instructors;
+
+		// Restriction action.
+		$data['restriction_action'] = $membership->get( 'restriction_redirect_type' );
+		$data['restriction_action'] = $data['restriction_action'] ? $data['restriction_action'] : 'none';
+
+		// Restriction message.
+		$data['restriction_message'] = array(
+			'raw'      => $membership->get( 'restriction_notice', $raw = true ),
+			'rendered' => do_shortcode( $membership->get( 'restriction_notice' ) ),
+		);
+		if ( empty( $data['restriction_message']['raw'] ) ) {
+			$data['restriction_message']['raw']      = 'You must belong to the [lifterlms_membership_link id="' .
+			                                           $membership->get( 'id' ) .
+			                                           '"] membership to access this content.';
+			$data['restriction_message']['rendered'] = do_shortcode( $data['restriction_message']['raw'] );
+		}
+
+		// Restriction page id.
+		if ( 'page' === $data['restriction_action'] || 'edit' === $context ) {
```

Comment:
I think you're totally right here :D We should always provide the `restriction_page_id`, `restriction_url`, `sales_page_page_id` and `sales_page_url` in edit context!!! Mostly because if someone wants to update them they must know the original value otherwise an error will be raised.
Thanks for implicitly highlighting this, I need to make a PR to align other resources too.